drivers/chown_unix.go

```diff
@@ -41,13 +41,21 @@ func (c *platformChowner) LChown(path string, info os.FileInfo, toHost, toContai
 		Ino: uint64(st.Ino),
 	}
 	c.mutex.Lock()
-	_, found := c.inodes[i]
+	oldOwner, found := c.inodes[i]
 	if !found {
-		c.inodes[i] = true
+		owner := idtools.IDPair{
+			UID: int(st.Uid),
+			GID: int(st.Gid),
+		}
+		c.inodes[i] = owner
 	}
 	c.mutex.Unlock()
 
-	if found {
+	// If the dev/inode was already chowned but the file has the original
+	// owner then it means the underlying hard link was broken (as it happens with
+	// overlay when index=off) so we need to chown again and not rely on the previous
+	// chown for the same inode.
```

Comment:
This comment really confused me - it didn't describe the checks on the next line, and after suggesting that if the checks were true, we still needed to do something, it just `return`ed.